src/extend-utils.ts

```diff
@@ -39,17 +26,23 @@ export async function getContractFactoryUtil (hre: HardhatRuntimeEnvironment, co
         throw new HardhatPluginError(PLUGIN_NAME, `Could not find ABI for ${contractPath}`);
     }
 
+    let gateway;
     const testNetworkName = hre.config.mocha.starknetNetwork || DEFAULT_STARKNET_NETWORK;
-
-    const network = getNetwork(testNetworkName, hre, "mocha.starknetNetwork");
     hre.starknet.network = testNetworkName;
```

Comment:
Does it make sense to set `hre.starknet.network` if it's not the one being used. This is possible in case networkURL is provided. Would it make more sense to have networkName as an argument instead of networkURL? Or the hre.starknet.network setting could be done outside this function?